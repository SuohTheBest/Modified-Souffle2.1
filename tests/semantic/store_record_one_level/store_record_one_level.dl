// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// Test storing of record with a single level of nesting

.type pair = [ first : number, second : number ]

.decl A(x: pair)
A([1, 2]).
A([1, -1]).
A([3, 3]).
A(nil).

.output A(IO=stdout,headers=true)
