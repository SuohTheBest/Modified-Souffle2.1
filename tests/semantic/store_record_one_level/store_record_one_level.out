---------------
A
x
===============
nil
[1, 2]
[1, -1]
[3, 3]
===============
